audit_events.go

```diff
@@ -0,0 +1,198 @@
+package gitlab
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"time"
+)
+
+// AuditEvent represents an audit event for a group or project
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ee/api/audit_events.html
+type AuditEvent struct {
+	ID         int               `json:"id"`
+	AuthorID   int               `json:"author_id"`
+	EntityID   int               `json:"entity_id"`
+	EntityType string            `json:"entity_type"`
+	Details    AuditEventDetails `json:"details"`
+	CreatedAt  *time.Time        `json:"created_at"`
+}
+
+// AuditEvent represents the details portion of an audit event for a group or project
+// The exact fields that are returned for an audit event depend on the action being recorded
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ee/api/audit_events.html
+type AuditEventDetails struct {
+	With          string      `json:"with"`
+	Add           string      `json:"add"`
+	As            string      `json:"as"`
+	Change        string      `json:"change"`
+	From          string      `json:"from"`
+	To            string      `json:"to"`
+	Remove        string      `json:"remove"`
+	CustomMessage string      `json:"custom_message"`
+	AuthorName    string      `json:"author_name"`
+	TargetID      TargetID    `json:"target_id"` // Sometimes this is an int and sometimes is a string with project path
```

Comment:
I prefer to use this field as an `interface{}` is this case. That seems to be more inline with the other locations were an ID that can be both a string and an integer.